### DISCUSSION TITLE
S1 Pro Compile Error

### DISCUSSION THREAD

**Discussion by vash2695** [mriscoc/Ender3V2S1#442] [Q&A] [answered]
I'm trying to compile FW for the F4 board on my S1 Pro and am having a couple of issues. I'm still pretty new to this so any help would be appreciated!

- Is the memory insufficient to enable both UBL and linear advance? 
- If so, are there any default options that can be disabled to free up enough space?
- Even with only linear advance enabled, I swapped the stock hot end fan and am trying to lower the default speed for Fan 1 to a value other than 255 but even in that case it overflowed by 788 bytes. To do this I attempted to change EXTRUDER_AUTO_FAN_SPEED. Aside from changing this value is there any way to set the speed of the extruder fan header elsewhere?

**Comment by mriscoc** [comment, marked as answer]
https://github.com/mriscoc/Ender3V2S1/discussions/407#discussioncomment-3789195

**Reply by vash2695** [reply]
That did it, thank you!
